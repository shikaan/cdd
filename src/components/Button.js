import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const getColorFromContext = (context, theme) => {
  if (context === CONTEXT.ACCENT) {
    return `
      background-color: ${theme.color.brand400};
      color: ${theme.color.light}
    `;
  } else {
    return null;
  }
};

export const CONTEXT = {
  ACCENT: "accent"
};

const StyledButton = styled.button(({theme, context}) => `
  font-family: ${theme.typography.primaryFont.fontFamily};
  font-size: ${theme.typography.baseFontSize.multiply(0.875)};
  font-weight: 700;
  border:none;
  padding: ${theme.typography.baseFontSize.multiply(0.5)} ${theme.typography.baseFontSize.multiply(1.5)};
  border-radius: 2px;

  ${getColorFromContext(context, theme)}  
`);

const Button = (props) => <StyledButton {...props} />;

Button.propTypes = {
  context: PropTypes.oneOf(Object.values(CONTEXT)),
  onClick: PropTypes.func
};

export default Button;
