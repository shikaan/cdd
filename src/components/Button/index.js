import React from "react";

export default class Button extends React.Component {
  render() {
    const {children, onClick} = this.props

    return <button onClick={onClick}>{children}</button>
  }
}
