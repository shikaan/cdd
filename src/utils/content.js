/**
 * @param {Array<{reference: string}>} microcopy
 * @param {string} key
 * @return {string}
 */
export const getMicrocopy = (microcopy, key) => {
  const element = microcopy?.find(i => i.reference === key) ?? {};

  return element.value ?? "";
};

/**
 * @param {Array<{reference: string}>} sections
 * @param {string} key
 * @return {string}
 */
export const getSection = (sections, key) => {
  return sections?.find(i => i.reference === key) ?? {reference: key};
};