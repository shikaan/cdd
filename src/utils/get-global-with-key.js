export const getGlobalWithKey = (key = "location") => {
  return typeof window !== "undefined" ? window : {[key]: {}};
};
