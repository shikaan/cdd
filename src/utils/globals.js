export const getGlobalWithKey = (key = "location") => {
  return typeof window !== "undefined" ? window : {[key]: {}};
};

export const getGlobal = () => { 
  // eslint-disable-next-line no-restricted-globals
  if (typeof self !== "undefined") { return self }
  if (typeof window !== "undefined") { return window } 
  if (typeof global !== "undefined") { return global } 
  throw new Error("unable to locate global object"); 
}; 
