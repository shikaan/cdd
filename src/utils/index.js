export {isLastIndex} from "./is-last-index";
export {getGlobalWithKey} from "./get-global-with-key";
export {isMobile, isTablet, isDesktop} from "./breakpoint";
export {validateSlug} from "./validate-slug";
export {getMicrocopy, getSection} from "./content";