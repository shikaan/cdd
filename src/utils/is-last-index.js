export const isLastIndex = (list, index) => {
  if (!Array.isArray(list)) {
    return false;
  }

  return index === list.length - 1;
};
