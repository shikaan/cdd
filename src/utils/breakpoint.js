import {numberBreakpoint} from "~theme/variables";
import {getGlobalWithKey} from "~/utils/get-global-with-key";

export const getViewportWidth = () => getGlobalWithKey("innerWidth") ?? getGlobalWithKey("document").documentElement.clientWidth;

export const isMobile = () => {
  const viewportWidth = getViewportWidth();
  return viewportWidth < numberBreakpoint.sm;
};

export const isTablet = () => {
  const viewportWidth = getViewportWidth();
  return numberBreakpoint.sm <= viewportWidth && viewportWidth < numberBreakpoint.md;
};

export const isDesktop = () => {
  const viewportWidth = getViewportWidth();
  return numberBreakpoint.md <= viewportWidth;
};