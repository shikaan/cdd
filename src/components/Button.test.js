import React from 'react'
import 'jest-styled-components';

import {shallowWithTheme} from "/test/utils";
import themeMock from '~theme'

import Button, {CONTEXT} from "./Button";

describe('Button', () => {
  it('renders correctly', () => {
    const wrapper = shallowWithTheme(<Button/>, themeMock)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with context', () => {
    const wrapper = shallowWithTheme(<Button context={CONTEXT.COFFEE}/>, themeMock)

    expect(wrapper).toMatchSnapshot()
  })
})
