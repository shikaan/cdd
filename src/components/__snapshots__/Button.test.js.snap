// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "breakpoint": Object {
        "lg": "1184px",
        "md": "980px",
        "mg": "1920px",
        "sm": "768px",
        "xl": "1366px",
        "xs": "480px",
      },
      "color": Object {
        "brand100": "hsla(23,92%,95%,1)",
        "brand200": "hsla(23,92%,82%,1)",
        "brand300": "hsla(23,92%,67%,1)",
        "brand400": "hsla(23,92%,52%,1)",
        "brand500": "hsla(23,92%,35%,1)",
        "brand600": "hsla(23,92%,18%,1)",
        "dark100": "hsla(44,8%,95%,1)",
        "dark200": "hsla(44,8%,82%,1)",
        "dark300": "hsla(44,8%,67%,1)",
        "dark400": "hsla(44,8%,52%,1)",
        "dark500": "hsla(44,8%,35%,1)",
        "dark600": "hsla(44,8%,18%,1)",
        "light": "hsla(0,100%,100%, 1)",
        "withTransparency": Object {
          "brand100": "hsla(23,92%,95%,.5)",
          "brand200": "hsla(23,92%,82%,.5)",
          "brand300": "hsla(23,92%,67%,.5)",
          "brand400": "hsla(23,92%,52%,.5)",
          "brand500": "hsla(23,92%,35%,.5)",
          "brand600": "hsla(23,92%,18%,.5)",
          "dark100": "hsla(44,8%,95%,.5)",
          "dark200": "hsla(44,8%,82%,.5)",
          "dark300": "hsla(44,8%,67%,.5)",
          "dark400": "hsla(44,8%,52%,.5)",
          "dark500": "hsla(44,8%,35%,.5)",
          "dark600": "hsla(44,8%,18%,.5)",
          "light": "hsla(0,100%,100%, .5)",
        },
      },
      "typography": Object {
        "__esModule": true,
        "baseFontSize": Size {
          "_rank": 2,
        },
        "baseLineHeight": "normal",
        "primaryFont": Object {
          "fontFamily": "Merriweather, Georgia, serif",
          "hover": "hsla(44,8%,18%,1)",
        },
        "secondaryFont": Object {
          "fontFamily": "Lato, Geneva, Tahoma, sans-serif",
          "hover": "hsla(44,8%,18%,1)",
        },
      },
      "zIndexRank": Object {
        "0": "0",
        "1": "100",
        "2": "200",
        "3": "300",
        "4": "400",
        "5": "500",
        "6": "600",
        "7": "700",
        "8": "800",
        "9": "900",
        "bottom": "-100",
        "top": "1000",
      },
    }
  }
>
  <Button />
</ContextProvider>
`;

exports[`Button renders correctly with context 1`] = `
<ContextProvider
  value={
    Object {
      "breakpoint": Object {
        "lg": "1184px",
        "md": "980px",
        "mg": "1920px",
        "sm": "768px",
        "xl": "1366px",
        "xs": "480px",
      },
      "color": Object {
        "brand100": "hsla(23,92%,95%,1)",
        "brand200": "hsla(23,92%,82%,1)",
        "brand300": "hsla(23,92%,67%,1)",
        "brand400": "hsla(23,92%,52%,1)",
        "brand500": "hsla(23,92%,35%,1)",
        "brand600": "hsla(23,92%,18%,1)",
        "dark100": "hsla(44,8%,95%,1)",
        "dark200": "hsla(44,8%,82%,1)",
        "dark300": "hsla(44,8%,67%,1)",
        "dark400": "hsla(44,8%,52%,1)",
        "dark500": "hsla(44,8%,35%,1)",
        "dark600": "hsla(44,8%,18%,1)",
        "light": "hsla(0,100%,100%, 1)",
        "withTransparency": Object {
          "brand100": "hsla(23,92%,95%,.5)",
          "brand200": "hsla(23,92%,82%,.5)",
          "brand300": "hsla(23,92%,67%,.5)",
          "brand400": "hsla(23,92%,52%,.5)",
          "brand500": "hsla(23,92%,35%,.5)",
          "brand600": "hsla(23,92%,18%,.5)",
          "dark100": "hsla(44,8%,95%,.5)",
          "dark200": "hsla(44,8%,82%,.5)",
          "dark300": "hsla(44,8%,67%,.5)",
          "dark400": "hsla(44,8%,52%,.5)",
          "dark500": "hsla(44,8%,35%,.5)",
          "dark600": "hsla(44,8%,18%,.5)",
          "light": "hsla(0,100%,100%, .5)",
        },
      },
      "typography": Object {
        "__esModule": true,
        "baseFontSize": Size {
          "_rank": 2,
        },
        "baseLineHeight": "normal",
        "primaryFont": Object {
          "fontFamily": "Merriweather, Georgia, serif",
          "hover": "hsla(44,8%,18%,1)",
        },
        "secondaryFont": Object {
          "fontFamily": "Lato, Geneva, Tahoma, sans-serif",
          "hover": "hsla(44,8%,18%,1)",
        },
      },
      "zIndexRank": Object {
        "0": "0",
        "1": "100",
        "2": "200",
        "3": "300",
        "4": "400",
        "5": "500",
        "6": "600",
        "7": "700",
        "8": "800",
        "9": "900",
        "bottom": "-100",
        "top": "1000",
      },
    }
  }
>
  <Button
    context="accent"
  />
</ContextProvider>
`;
